rdrf/rdrf/views/form_view.py

```diff
@@ -359,6 +360,30 @@ def delete(self, request, registry_code, form_id, patient_id, context_id=None):
 
         return JsonResponse({"result": "Cannot delete form !"}, status=400)
 
+    def has_valid_randomised_values(self, request, dynamic_data):
+
+        def verify_value(random_value, dynamic_data):
+            if isinstance(random_value, list):
+                if set(random_value) != set(dynamic_data[field_name]):
+                    logger.info(f"Invalid randomised value for {field_name}: {dynamic_data[field_name]} != {random_value}")
+                    return False
+            elif random_value != dynamic_data[field_name]:
+                logger.info(f"Invalid randomised value for {field_name}: {dynamic_data[field_name]} != {random_value}")
+                return False
+            return True
+
```

Comment:
I think these checks are the other way around.

The code assumes that the values are in the session and loops over them to check they match the values received.
If the user POSTs without going to the GET page first there won't be any values in the session and they could send in any values they want to set.